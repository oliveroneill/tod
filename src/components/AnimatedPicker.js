import React, { Component } from 'react';
import {
  Dimensions,
  StyleSheet,
  Text,
  View,
  DatePickerIOS,
  Picker,
  Animated,
  TouchableHighlight,
  Platform,
} from 'react-native';
import DateTimePicker from 'react-native-modal-datetime-picker';
import Modal from 'react-native-modal';

var deviceWidth = Dimensions.get('window').width;
var deviceHeight = Dimensions.get('window').height;
var PickerItem = Picker.Item;

const styles = StyleSheet.create({
  closeButtonContainer: {
    flexDirection: 'row',
    justifyContent: 'flex-end',
    borderTopColor: '#e2e2e2',
    borderTopWidth: 1,
    borderBottomWidth:1,
    borderBottomColor: '#e2e2e2',
    backgroundColor: '#ffffff',
    paddingTop:10,
    paddingBottom:10
  },
  closeButton: {
    paddingRight:10,
  },
  closeButtonText: {
    color: '#027afe'
  }
});
class AnimatedPicker extends Component {
  usingAndroidDatePicker() {
    return Platform.OS === 'android' && this.props.type === "date" && this.props.isVisible;
  }
  render() {
    var defaultDate = new Date();
    if (this.props.type === 'date')
      defaultDate = this.props.currentOption;
    return (
      <View>
        <DateTimePicker
          isVisible={this.usingAndroidDatePicker()}
          onConfirm={this.props.onOptionChange}
          onCancel={this.props.closeModal}
          mode='datetime'
          date={defaultDate}
          minimumDate={this.props.minDate}
          maximumDate={this.props.maxDate}
        />
        <Modal
          isVisible={this.props.isVisible && !this.usingAndroidDatePicker()}
          style={{justifyContent:'flex-end', margin:0}}
          backdropOpacity={0}
        >
          <View style={styles.closeButtonContainer}>
            <TouchableHighlight
              onPress={ this.props.closeModal}
              underlayColor="transparent"
              style={styles.closeButton}
            >
              <Text style={styles.closeButtonText}>Done</Text>
            </TouchableHighlight>
          </View>
          {this.props.type === "date" ?
            <DatePickerIOS
              date={this.props.currentOption}
              minimumDate={this.props.minDate}
              maximumDate={this.props.maxDate}
              mode="datetime"
              onDateChange={this.props.onOptionChange}
            />
          :
            <Picker
              style={{backgroundColor:'#ffffff'}}
              selectedValue={this.props.currentOption}
              onValueChange={this.props.onOptionChange}>
              {Array.isArray(this.props.options) ?
                this.props.options.map((option) => (
                  <PickerItem
                    key={option}
                    value={option}
                    label={option}
                  />
                ))
                :
                Object.keys(this.props.options).map((key) => (
                  <PickerItem
                    key={key}
                    value={key}
                    label={this.props.options[key]}
                  />
                ))
              }
            </Picker>
          }
        </Modal>
      </View>
    )
  }
}

export default AnimatedPicker;
