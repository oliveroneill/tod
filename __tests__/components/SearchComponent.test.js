import React from 'react';
import renderer from 'react-test-renderer';

import {
  Text,
  TouchableHighlight,
  TextInput,
  ListView
} from 'react-native';
import { ListItem } from 'react-native-elements'
import moment from 'moment-timezone';

import { shallow } from 'enzyme';

import SearchComponent from '../../src/components/SearchComponent.js';
import AnimatedPicker from '../../src/components/AnimatedPicker.js';
import IconRow from '../../src/components/IconRow.js';

function createMockProps(currentLocation, destination) {
  return {
    navigation: {
      setParams: jest.fn(),
      state: {
        params: destination
      },
      navigate: jest.fn()
    },
    currentLocation: currentLocation,
    api: {
      getRoutes: jest.fn().mockReturnValue(
        new Promise(function(resolve, reject) {})
      )
    }
  };
}

function setup(lat, lng) {
  let props = createMockProps({lat, lng});
  return shallow(<SearchComponent {...props} />);
}

describe('SearchComponent', () => {
  it('displays views correctly', () => {
    // snapshot test
    let lat = -44;
    let lng = 22;
    let props = createMockProps({lat, lng});
    let component = renderer.create(<SearchComponent {...props} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('requests the search input', () => {
    // given
    let origin = {lat: -23, lng: 25};
    let destination = {
      name: 'Test Address Name, City, Country',
      lat: 21,
      lng: 90
    };
    let props = createMockProps(origin, destination);
    // when
    let enzymeWrapper = shallow(<SearchComponent {...props} />);
    // set transport type
    let selectedIndex = 2;
    enzymeWrapper.find(IconRow).props().onPress(selectedIndex);
    enzymeWrapper.update();
    // set arrival time
    enzymeWrapper.find(TouchableHighlight).first().props().onPress();
    enzymeWrapper.update();
    // clear previous queries
    props.api.getRoutes.mockClear();
    let arrival = 1482363397071;
    enzymeWrapper.find(AnimatedPicker).props().onOptionChange(new Date(arrival));
    enzymeWrapper.update();
    // then
    expect(props.api.getRoutes).toHaveBeenCalledWith(origin, destination, "walking", arrival);
  });

  it('displays routes correctly', () => {
    // snapshot test
    let lat = -44;
    let lng = 22;
    let props = createMockProps({lat, lng});
    let screen = new SearchComponent({...props});
    let mockRowData = {
      arrival_time: 1482363467071,
      departure_time: 1482363367071,
      description: "Take x avenue",
      name: "Bus Route 1"
    };
    let component = renderer.create(screen.renderCell(mockRowData, 1));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('updates routes on input change', () => {
    // given
    let origin = {lat: -23, lng: 25};
    let destination = {
      name: 'Test Address Name, City, Country',
      lat: 21,
      lng: 90
    };
    let props = createMockProps(origin, destination);
    let enzymeWrapper = shallow(<SearchComponent {...props} />);
    // set transport type
    let selectedIndex = 2;
    enzymeWrapper.find(IconRow).props().onPress(selectedIndex);
    enzymeWrapper.update();
    // set arrival time
    enzymeWrapper.find(TouchableHighlight).first().props().onPress();
    enzymeWrapper.update();
    // clear previous queries
    props.api.getRoutes.mockClear();
    var arrival = 1482363397071;
    enzymeWrapper.find(AnimatedPicker).props().onOptionChange(new Date(arrival));
    enzymeWrapper.update();
    expect(props.api.getRoutes).toHaveBeenCalledWith(origin, destination, "walking", arrival);
    // update arrival time and ensure that a new query is made
    props.api.getRoutes.mockClear();
    arrival = 1482362397071;
    enzymeWrapper.find(AnimatedPicker).props().onOptionChange(new Date(arrival));
    enzymeWrapper.update();
    expect(props.api.getRoutes).toHaveBeenCalledWith(origin, destination, "walking", arrival);
  });

  it('displays error when route search fails', () => {
    // given
    let origin = {lat: -23, lng: 25};
    let destination = {
      name: 'Test Address Name, City, Country',
      lat: 21,
      lng: 90
    };
    // will fail on API call
    let promise = Promise.reject();
    let props = createMockProps(origin, destination);
    props.api.getRoutes = jest.fn().mockReturnValue(promise);
    // when
    let wrapper = shallow(<SearchComponent {...props} />);
    wrapper.find(IconRow).props().onPress(2);
    wrapper.update();
    // must wait until after the error was received by the component
    return promise
    .catch(() => {})
    .then(() => {
      // then
      wrapper.update();
      let expected = "Routes (Something went wrong):";
      expect(wrapper.find(Text).at(3).props().children.join("")).toEqual(expected);
    });
  });

  it('displays loading message when searching', () => {
    // given
    let origin = {lat: -23, lng: 25};
    let destination = {
      name: 'Test Address Name, City, Country',
      lat: 21,
      lng: 90
    };
    let props = createMockProps(origin, destination);
    props.api.getRoutes = jest.fn().mockReturnValue(Promise.reject());
    // when
    let wrapper = shallow(<SearchComponent {...props} />);
    // switch transport modes
    wrapper.find(IconRow).props().onPress(2);
    wrapper.update();
    // then
    let expected = "Routes (Loading):";
    expect(wrapper.find(Text).at(3).props().children.join("")).toEqual(expected);
  });

  it('navigates to schedule screen on press', () => {
    // given
    let lat = -44;
    let lng = 22;
    let destination = {
      name: 'Test Address Name, City, Country',
      lat: 21,
      lng: 90
    };
    let props = createMockProps({lat, lng});
    let screen = new SearchComponent({...props});
    let mockRowData = {
      arrival_time: 1482363467071,
      departure_time: 1482363367071,
      description: "Take x avenue",
      name: "Bus Route 1"
    };
    let inputArrival = 1482383467071;
    let mockTimezone = "America/Los_Angeles";
    moment.tz.guess = jest.fn().mockReturnValue(mockTimezone);
    let waitingWindow = 2;
    screen.state = {
      waitingWindow: waitingWindow,
      date: new Date(inputArrival),
      destination: destination,
      transportIndex: 1
    };
    // when
    let wrapper = shallow(screen.renderCell(mockRowData, 1));
    wrapper.props().onPress();
    // then
    // convert waiting window to milliseconds
    let expectedWaitingWindow = waitingWindow * 60 * 1000;
    let expected = {
      origin: {lat, lng },
      dest: destination,
      route:mockRowData,
      inputArrival: inputArrival,
      inputArrivalDateString:"2016-12-21T21:11:07-08:00 PST",
      departure:mockRowData.departure_time - expectedWaitingWindow,
      subtitle:"Take x avenue - Alert at 5:34 pm and arrive at 5:37 pm",
      transportIcon: "directions-car",
      transport: "driving",
      waitingWindowMs: expectedWaitingWindow,
      timezoneLocation: mockTimezone,
      api: props.api
    };
    expect(props.navigation.navigate).toHaveBeenCalledWith('AddAlert', expected);
  });
});
