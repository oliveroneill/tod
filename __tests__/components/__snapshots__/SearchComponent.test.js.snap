// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent displays routes correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderBottomColor": "#bbb",
          "borderBottomWidth": 1,
          "paddingBottom": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
        undefined,
        false,
        false,
      ],
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginLeft": 10,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontSize": 17.5,
                },
                Object {
                  "marginLeft": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          Bus Route 1
        </Text>
      </View>
      <View
        style={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={3}
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#86939e",
                  "fontSize": 15,
                  "fontWeight": "600",
                  "marginTop": 1,
                },
                Object {
                  "marginLeft": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        >
          Take x avenue - Alert at 5:31 pm and arrive at 5:37 pm
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        onPress={undefined}
        style={
          Array [
            false,
            false,
            false,
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
        underlayColor="white"
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bdc6cf",
                "fontSize": 28,
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ],
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`SearchComponent displays views correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#e1e8ee",
        "borderBottomColor": "#e1e1e1",
        "borderBottomWidth": 1,
        "borderTopColor": "#e1e1e1",
        "borderTopWidth": 1,
        "height": 50,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onFocus={[Function]}
      placeholder="Destination"
      placeholderTextColor="#86939e"
      style={
        Object {
          "backgroundColor": "#bdc6cf",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 13,
          "height": 30,
          "margin": 8,
          "marginLeft": 8,
          "marginRight": 8,
          "marginTop": 7.5,
          "overflow": "hidden",
          "paddingBottom": 4.5,
          "paddingLeft": 26,
          "paddingRight": 19,
          "paddingTop": 4.5,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "margin": 7,
            },
            Object {
              "borderRadius": 34,
              "height": 64,
              "width": 64,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#00aced",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "margin": 7,
            },
            Object {
              "borderRadius": 34,
              "height": 64,
              "width": 64,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#bdc6cf",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "margin": 7,
            },
            Object {
              "borderRadius": 34,
              "height": 64,
              "width": 64,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#bdc6cf",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Array [
            Object {
              "margin": 7,
            },
            Object {
              "borderRadius": 34,
              "height": 64,
              "width": 64,
            },
            Object {
              "shadowColor": "rgba(0,0,0, .4)",
              "shadowOffset": Object {
                "height": 1,
                "width": 1,
              },
              "shadowOpacity": 1,
              "shadowRadius": 1,
            },
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "justifyContent": "center",
            },
            undefined,
          ],
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 30,
            },
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "color": "#bdc6cf",
              },
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "paddingBottom": 12,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 12,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 11,
            "textAlign": "center",
          }
        }
      >
        Arrive at: 
        Wednesday 6:06pm
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "paddingBottom": 12,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 12,
        }
      }
    >
      |
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "paddingBottom": 12,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 12,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 11,
          }
        }
      >
        Alert 
        5 mins
         before
      </Text>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "paddingLeft": 10,
      }
    }
  >
    Routes:
  </Text>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#bbb",
          "borderTopWidth": 1,
          "marginTop": 20,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": "unknown",
        }
      }
      enableEmptySections={true}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        View In Google Maps
      </Text>
    </View>
  </View>
  <View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
        testID={undefined}
      />
      <View
        collapsable={undefined}
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 10,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "borderRadius": 13,
                "marginBottom": 8,
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "transparent",
                "borderBottomColor": "#d5d5d5",
                "borderBottomWidth": 0.5,
                "padding": 14,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#8f8f8f",
                    "fontSize": 13,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Pick a date
            </Text>
          </View>
          <View
            onStartShouldSetResponderCapture={[Function]}
          >
            <View
              style={undefined}
            >
              <RCTDatePicker
                date={1482363367071}
                maximumDate={1490135767071}
                minimumDate={1482363367071}
                minuteInterval={undefined}
                mode="datetime"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 216,
                  }
                }
                timeZoneOffsetInMinutes={undefined}
              />
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "#d5d5d5",
                  "borderTopWidth": 0.5,
                  "height": 57,
                  "justifyContent": "center",
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#007ff9",
                    "fontSize": 20,
                    "fontWeight": "normal",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Confirm
            </Text>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "backgroundColor": "white",
                "borderRadius": 13,
                "height": 57,
                "justifyContent": "center",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "600",
                  "padding": 10,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </Modal>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
        testID={undefined}
      />
      <View
        collapsable={undefined}
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
            "margin": 0,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderBottomColor": "#e2e2e2",
              "borderBottomWidth": 1,
              "borderTopColor": "#e2e2e2",
              "borderTopWidth": 1,
              "flexDirection": "row",
              "justifyContent": "flex-end",
            }
          }
        >
          <View>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              hasTVPreferredFocus={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  },
                ]
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#027afe",
                    },
                    Object {
                      "color": "#027afe",
                    },
                  ]
                }
              >
                Done
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
            }
          }
        >
          <RCTPicker
            items={Array []}
            onChange={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </Modal>
  </View>
</View>
`;
